### ISSUE TITLE
Contacts page slow to load with 10M records - missing indices

### ISSUE THREAD

**Issue by riisi** [mautic/mautic#14369]
### Mautic Series

5.1.x series (not supported)

### Mautic installed version

5.1.1

### Way of installing

I cloned the Mautic repo (not recommended or supported)

### PHP version

8.1

### What browsers are you seeing the problem on?

Not relevant

### What happened?

I have been doing some load testing with increasing numbers of contacts. With 10M contacts loaded, observed that the contacts page did not load. 

The underlying database query was taking longer to execute than the cloud load balancer timeouts, as well as PHP max execution time and fast_cgi timeouts. After increasing the timeouts, the page loads in about 1 minute. 

The cause seems to be a slow db query getting the contacts data (per CloudSQL query insights): 
```sql
SELECT l.* FROM leads l WHERE l.date_identified IS NOT ? ORDER BY l.last_active DESC, l.id DESC LIMIT ? OFFSET ?
```
<img width="676" alt="image" src="https://github.com/user-attachments/assets/771a5eea-9026-4ad5-89a6-4fd6b4e58a8b" />

This query was taking ~40 seconds. This is using GCP CloudSQL Enterprise Plus DB with 4 CPU, 32 GB RAM.
This issue was not observed when testing with 1M contacts.

We can see this also in an application trace - 
<img width="848" alt="App trace" src="https://github.com/user-attachments/assets/e22703bd-bba0-4449-b408-342d7a4ee753" />

Two queries are executed - one to get the total count (`SELECT COUNT ( l.id ) FROM leads l WHERE l.date_identified IS NOT ?`) and the one above to get the records.

After noticing that the query does a full table scan, I added indices for `id` and `date_identified`. This reduced the total page load time to ~2s. 

It seems to me these indexes should be added. If it is a straightforward change I am also happy to try to help.

It's not clear to me why the `date_identified` filter is needed.

### How can we reproduce this issue?

Step 1: Load a large number of contacts in Mautic and observe query times with & without the indices.

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by RCheesley** [comment]
Thanks for the report @riisi - I've flagged this up in the performance and scalability channel on Slack for discussion. Would you be open to making a PR if this is something that the core team would accept?

**Comment by escopecz** [comment]
This is actually possible to do from the UI:
![Screenshot 2024-12-16 at 13 41 16](https://github.com/user-attachments/assets/9dcd825b-4b46-4595-80b9-09aa666736b3)

But I'm OK if the index exists by default.

> It's not clear to me why the date_identified filter is needed.

It's often used in segments where marketers usually want to exclude contacts that are not identified.

**Comment by escopecz** [comment]
I'm closing this issue as there was no response to my previous message so I assume it fixed the problem

**Comment by riisi** [comment]
Thanks @escopecz. I think this should still be fixed long term but lower priority given the UI. Mainly wanted to log this now for awareness.
